services/worker-manager/src/providers/provider.js

```diff
@@ -85,6 +85,20 @@ class Provider {
 
   async removeResources({workerPool}) {
   }
+
+  interpretLifecycle({lifecycle}) {
+    let registrationExpiry, reregisterDeadline = null;
+    if (lifecycle){
+      const {registrationTimeout, reregisterTimeout} = lifecycle;
+      if (registrationTimeout) {
+        registrationExpiry = Date.now() + registrationTimeout * 1000;
+      }
+      if (reregisterTimeout) {
+        reregisterDeadline = Date.now() + reregisterTimeout * 1000;
```

Comment:
The schema docs say this is the time since the `register()` call, but here it's being calculated when the instance is created. (this is the reason for the r-)